From 6d7ca47757e35bfc1478bda5f1f46d8cc90f0e67 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 55/70] media: fix infinite wait at source for HAL1 based
 recording

if recording is stopped by pressing home key,
camera immediatly stops giving the frames as
surface is destroyed. Puller could not call
setStopTime as puller is stuck at read with
camerasource's infinite wait for video frame.
Calling setStopTime immediatly to avoid deadlock.
---
 media/libstagefright/MediaCodecSource.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/media/libstagefright/MediaCodecSource.cpp b/media/libstagefright/MediaCodecSource.cpp
index aff612d7e..0bd5c4c29 100644
--- a/media/libstagefright/MediaCodecSource.cpp
+++ b/media/libstagefright/MediaCodecSource.cpp
@@ -167,9 +167,7 @@ status_t MediaCodecSource::Puller::postSynchronouslyAndReturnError(
 }
 
 status_t MediaCodecSource::Puller::setStopTimeUs(int64_t stopTimeUs) {
-    sp<AMessage> msg = new AMessage(kWhatSetStopTimeUs, this);
-    msg->setInt64("stop-time-us", stopTimeUs);
-    return postSynchronouslyAndReturnError(msg);
+    return mSource->setStopTimeUs(stopTimeUs);
 }
 
 status_t MediaCodecSource::Puller::start(const sp<MetaData> &meta, const sp<AMessage> &notify) {
-- 
2.11.0

